### PR TITLE
Add parse list test

### PR DESCRIPTION
**Pull request by benson31**
Adds the tests I should have added as part of #1118.

### CODE REVIEW

File: src/proto/unit_test/parse_list_test.cpp [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    CHECK(answer == expected);
+    CHECK(
+      lbann::parse_list<lbann::execution_mode>("")
+      == std::vector<lbann::execution_mode>{});
+    CHECK(
+      lbann::parse_list<lbann::execution_mode>(" ")
+      == std::vector<lbann::execution_mode>{});
+
+    CHECK_THROWS(
+      lbann::parse_list<lbann::execution_mode>("banana tuna salad"));
```
Comment (timmoon10) [inline src/proto/unit_test/parse_list_test.cpp]: Sounds gross.

File: src/proto/unit_test/parse_list_test.cpp [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+      lbann::parse_list<std::string>("this is a test");
+    CHECK(answer == expected);
+    CHECK(
+      lbann::parse_list<std::string>("") == std::vector<std::string>{});
+
+  }
+
+  SECTION("int")
+  {
+    const std::vector<int> expected = { 1, 2, 3, 4, 5};
```
Comment (timmoon10) [inline src/proto/unit_test/parse_list_test.cpp]:
```suggestion
    const std::vector<int> expected = { 1, 2, 3, 4, 5 };
```

File: src/proto/unit_test/parse_set_test.cpp [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    const std::set<std::string> expected = { "this", "is", "a", "test" };
+    auto const answer =
+      lbann::parse_set<std::string>("this is a test");
+    CHECK(answer == expected);
+    CHECK(lbann::parse_set<std::string>("") == std::set<std::string>{});
+  }
+
+  SECTION("int")
+  {
+    const std::set<int> expected = { 1, 2, 3, 4, 5};
```
Comment (timmoon10) [inline src/proto/unit_test/parse_set_test.cpp]:
```suggestion
    const std::set<int> expected = { 1, 2, 3, 4, 5 };
```

File: src/proto/unit_test/parse_list_test.cpp (hunk 1 above)
Comment (benson31) [reply src/proto/unit_test/parse_list_test.cpp]: Yep, that's why I picked it. Should I
```cpp
#define CHECK_THROWS_UP(...) \
    CHECK_THROWS(__VA_ARGS__)
```
to help drive the point home? 😛

**Pull request merged**